controllers/storageclaim_controller.go

```diff
@@ -368,99 +420,77 @@ func (r *StorageClassClaimReconciler) reconcilePhases() (reconcile.Result, error
 				data["csi.storage.k8s.io/snapshotter-secret-namespace"] = r.storageClient.Namespace
 				// generate a new clusterID for cephfs subvolumegroup, as
-				// storageclassclaim is clusterscoped resources using its
+				// storageclaim is clusterscoped resources using its
 				// name as the clusterID
-				data["clusterID"] = r.storageClassClaim.Name
+				data["clusterID"] = r.storageClaim.Name
 				if resource.Name == "cephfs" {
 					volumeSnapshotClass = r.getCephFSVolumeSnapshotClass(data)
 				} else if resource.Name == "ceph-rbd" {
 					volumeSnapshotClass = r.getCephRBDVolumeSnapshotClass(data)
 				}
-				utils.AddAnnotation(volumeSnapshotClass, storageClassClaimAnnotation, r.storageClassClaim.Name)
+				utils.AddAnnotation(volumeSnapshotClass, storageClaimAnnotation, r.storageClaim.Name)
 				if err := r.createOrReplaceVolumeSnapshotClass(volumeSnapshotClass); err != nil {
 					return reconcile.Result{}, fmt.Errorf("failed to create or update VolumeSnapshotClass: %s", err)
 				}
 			}
 		}
 
 		// update monitor configuration for cephcsi
-		err = cc.UpdateMonConfigMap(r.storageClassClaim.Name, r.storageClient.Status.ConsumerID, csiClusterConfigEntry)
+		err = cc.UpdateMonConfigMap(r.storageClaim.Name, r.storageClient.Status.ConsumerID, csiClusterConfigEntry)
 		if err != nil {
 			return reconcile.Result{}, fmt.Errorf("failed to update mon configmap: %v", err)
 		}
 		// Readiness phase.
-		// Update the StorageClassClaim status.
-		r.storageClassClaim.Status.Phase = v1alpha1.StorageClassClaimReady
+		// Update the StorageClaim status.
+		r.storageClaim.Status.Phase = v1alpha1.StorageClaimReady
 
-		// Initiate deletion phase if the StorageClassClaim exists.
-	} else if r.storageClassClaim.UID != "" {
+		// Initiate deletion phase if the StorageClaim exists.
+	} else if r.storageClaim.UID != "" {
 		// Deletion phase.
-		// Update the StorageClassClaim status.
-		r.storageClassClaim.Status.Phase = v1alpha1.StorageClassClaimDeleting
-
-		// Delete StorageClass.
-		// Make sure there are no StorageClass consumers left.
-		// Check if StorageClass is in use, if yes, then fail.
-		// Wait until all PVs using the StorageClass under deletion are removed.
-		// Check for any PVs using the StorageClass.
-		pvList := corev1.PersistentVolumeList{}
-		err := r.list(&pvList)
-		if err != nil {
-			return reconcile.Result{}, fmt.Errorf("failed to list PersistentVolumes: %s", err)
-		}
-		for i := range pvList.Items {
-			pv := &pvList.Items[i]
-			if pv.Spec.StorageClassName == r.storageClassClaim.Name {
-				return reconcile.Result{}, fmt.Errorf("StorageClass %s is still in use by one or more PV(s)",
-					r.storageClassClaim.Name)
-			}
+		// Update the StorageClaim status.
+		r.storageClaim.Status.Phase = v1alpha1.StorageClaimDeleting
+
+		if exist, err := r.hasStorageClaimDependents(); err != nil {
+			return reconcile.Result{}, fmt.Errorf("failed to verify storageclaim dependent resources: %v", err)
```

Comment:
This message indicates that we will not delete if we have any type of owned resource that is incorrect. Can we fix the message

Reply:
sry, didn't get the ask. By existing msg I mean to say, we couldn't decide on the deletion, probably next reconcile might fix it.

Reply:
No, just that the message is misleading

Reply:
refactored. done.